recorder_test.go

```diff
@@ -3,185 +3,285 @@ package lightstep
 import (
 	"encoding/json"
 	"fmt"
-	"reflect"
 	"strings"
-	"testing"
 	"time"
 
 	google_protobuf "github.com/golang/protobuf/ptypes/timestamp"
 	"github.com/lightstep/lightstep-tracer-go/basictracer"
 	cpb "github.com/lightstep/lightstep-tracer-go/collectorpb"
-	"github.com/lightstep/lightstep-tracer-go/thrift_rpc"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 	ot "github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/log"
 )
 
-const (
-	arbitraryTimestampSecs = 1473442150
-)
-
 func makeSpanSlice(length int) []basictracer.RawSpan {
 	return make([]basictracer.RawSpan, length)
 }
 
-func makeExpectedLogs() []*cpb.Log {
-	eRes := make([]*cpb.Log, 8)
-	for i := 0; i < 8; i++ {
-		pl, _ := json.Marshal([]interface{}{i, i, true, "suhhh"})
-		eRes[i] = &cpb.Log{
-			Timestamp: &google_protobuf.Timestamp{arbitraryTimestampSecs, 0},
-			Keyvalues: []*cpb.KeyValue{
-				&cpb.KeyValue{Key: "string", Value: &cpb.KeyValue_StringValue{fmt.Sprintf("foo%d", i)}},
-				&cpb.KeyValue{Key: "object", Value: &cpb.KeyValue_JsonValue{string(pl)}},
-				&cpb.KeyValue{
-					Key:   "too_long-----------…",
-					Value: &cpb.KeyValue_StringValue{"---------------------------------------…"},
-				},
-			},
-		}
-	}
-	return eRes
-}
+var _ = Describe("Recorder", func() {
+	const (
+		arbitraryTimestampSecs = 1473442150
```

Comment:
In Ginkgo, it's best to declare variables and constants as close to the tests in which they are used, for clarity and to avoid accidental misuse. Move arbitraryTimestampSecs and the other top-level vars down to the tightest scope that is reasonable (most are only used in the LogRecords tests).

Also, this is just a personal style, but I prefer the single line declaration pattern for var and const, rather than the block pattern, as it often saves space.

```go
	const arbitraryTimestampSecs = 1473442150

	var recorder  *Recorder
        var tracer    ot.Tracer
	var timestamp time.Time
	var logs      []*cpb.Log
```